lib/Frontend/CachingUtils.cpp

```diff
@@ -0,0 +1,31 @@
+//===--- CachingUtils.cpp ---------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Frontend/CachingUtils.h"
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/Path.h"
+
+using namespace llvm;
+
+namespace swift {
+
+std::string getDefaultSwiftCASPath() {
+  SmallString<256> Path;
+  if (!llvm::sys::path::cache_directory(Path))
+    llvm::report_fatal_error("cannot get default cache directory");
+  llvm::sys::path::append(Path, "swift-cache");
```

Comment:
I think we should just use clang's default path. While the object files won't overlap, many header files will get ingested by both clang and swift and I don't see an advantage to splitting them.

Reply:
Should we just fall back to use `llvm::cas::getDefaultOnDiskCASPath` and get rid of the clang version?

Reply:
Seems reasonable to me; @akyrtzi any objections to removing `clang::getClangDefaultCachePath` and just use `llvm::cas::getDefaultOnDiskCASPath`?

Reply:
SGTM.